### PR TITLE
Allow transparent plugin backgrounds in caption rows

### LINKED ISSUE #61: Allow "transparent" background colors
# Summary

Currently we are forced to choose a solid color for the backgrounds, and are unable to use `tiki` with the terminal's background itself (i.e. no background color applied by `tiki`).
It would be really great to have this option.

# Background

**Is your feature request related to a problem? Please describe**:
Yes. I have my terminal setup to be just the exact level of window transparency I need, and I like having things visible in the background. Applying a solid color on top destroys this.

**Describe the solution you'd like**:
Allow the `background` configs to take a keyword `transparent` (or whatever else you choose) as a way to denote no background color should be applied.

# Details

You can see the terminal's bg (which is slightly transparent) at the very top, can see firefox's address bar behind it.

Image

### PR DESCRIPTION
**Pull request by nickzerjeski**
## Summary
- support `transparent` / `none` / `default` / `inherit` color keywords in plugin color parsing
- keep plugin title-row backgrounds transparent when background is default
- preserve existing gradient rendering for non-transparent backgrounds

Closes #61

## Testing
- `go test ./plugin ./view ./view/taskdetail ./task ./model`

### CODE REVIEW

File: plugin/colorparser.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ func parseColor(s string, defaultColor tcell.Color) tcell.Color {
 	if s == "" {
 		return defaultColor
 	}
+	switch strings.ToLower(s) {
+	case "transparent", "none", "default", "inherit":
+		return tcell.ColorDefault
+	}
```
[the remark this answers is not in the document]
Comment (nickzerjeski) [reply plugin/colorparser.go:18]: Implemented in commit 9f19802: parser now tracks explicit field presence (`ForegroundSet`/`BackgroundSet`) and merge logic uses those flags instead of `IsDefault()`, so explicit transparent/default overrides are applied correctly.

File: plugin/colorparser_test.go [hunk 2]
```diff
@@ -68,6 +68,12 @@ func TestParseColor(t *testing.T) {
 			defaultColor: tcell.ColorWhite,
 			want:         tcell.GetColor("#ff0000"),
 		},
+		{
+			name:         "transparent keyword returns default color",
+			input:        "transparent",
+			defaultColor: tcell.ColorWhite,
+			want:         tcell.ColorDefault,
+		},
```
[the remark this answers is not in the document]
Comment (nickzerjeski) [reply plugin/colorparser_test.go:76]: Added in commit 9f19802: extra test cases now cover additional case-insensitive keywords/synonyms (`"NONE"`, and whitespace-trimmed `"default"`) to lock in behavior beyond only `"transparent"`.

**Comment by nickzerjeski** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/boolean-maybe/tiki/pull/62#pullrequestreview-4099168887)
